### DISCUSSION TITLE
Custom container

### DISCUSSION THREAD

**Discussion by Haw984** [InterDigitalInc/AdvantEDGE#438] [Ideas]
Hi,
I am new to this platform and would like to add a router container image to the topology. The available options are edge applications, cloud apps, or terminal apps, where I can add containers. I want to check if router container can be added on the 'ZONE' node.

Kind Regards.

**Comment by dilallkx** [comment]
Hi @HawUm,

Thank you for using our platform and reaching out!
Containers can only be configure in Processes, and Processes can only be added to physical nodes as described in the [AdvantEDGE Network Model](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/overview-architecture/#network-model). Other network elements are simply used to build a network topology and apply network characteristics to the end-to-end application traffic.

Optionally, you could create a zone with 0 latency, add an edge node to the zone and add an edge application to the edge node. This would be like having an edge application deployed directly in a Zone. It depends on the scenario topology you want to build and test.

Also, you mention wanting to deploy a router container image; please be careful that the container you deploy in your scenario does not affect the netfiler/iptable/traffic control rules enforced by the AdvantEDGE platform.

I hope this helps!
Best regards!
